lib/presentation/Form/bloc/myform_bloc.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:bloc/bloc.dart';
+
+import '../../../data/constants/app_assets.dart';
+import '../../../domain/repositories/user_repository.dart';
+import 'myform_event.dart';
+import 'myform_state.dart';
+
+class MyFormBloc extends Bloc<MyFormEvent, MyFormState> {
+  MyFormBloc({required this.userRepository, required bool isEditing})
+      : super(MyFormState(isEditing: isEditing)) {
+    on<MyFormSubmitEvent>((MyFormSubmitEvent event, Emitter<MyFormState> emit) {
+      try {
+        userRepository.addFormDetails(
+            event.formData[nameKey]!,
```

Comment:
again hope there is logic to asseting these as not null as if map doesnt contain that key it would crash